### PR TITLE
Add get user by role endpoint

### PR DESCRIPTION
**Pull request by jls-github**
### What issue is this solving?

codeforpdx/dwellingly-app/issues/296 (should not close - frontend work required)

Supplies backend api endpoint to display user data based on queried role (formatted '/user?r=<int>'). Endpoint is locked behind admin permissions. This will allow us to populate the JOIN staff page on the front end.

### Any helpful knowledge/context for the reviewer?
Is a re-seeding of the database necessary?
Any new dependencies to install?
Any special requirements to test?

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code

### CODE REVIEW

File: tests/integration/test_users.py [hunk 1]
```diff
@@ -197,4 +197,26 @@ def test_delete_user(client, auth_headers, new_user):
...
+def test_get_user(client, auth_headers, new_user):
+    """GET '/user' returns a list of all users queried by role"""
+
+    staff_user_response = client.get('/api/user?r=3', headers=auth_headers["admin"])
+    admin_user_response = client.get('/api/user?r=4', headers=auth_headers["admin"])
+
+    assert is_valid(staff_user_response, 200)
+    assert is_valid(admin_user_response, 200)
+    assert all(staff["role"] == 3 for staff in staff_user_response.json["users"])
+    assert all(admin["role"] == 4 for admin in admin_user_response.json["users"])
```
Comment (NickSchimek) [inline tests/integration/test_users.py]:
```suggestion
    assert all(staff["role"] == RoleEnum.STAFF.value for staff in staff_user_response.json["users"])
    assert all(admin["role"] == RoleEnum.ADMIN.value for admin in admin_user_response.json["users"])
```

File: tests/integration/test_users.py [hunk 2]
```diff
@@ -197,4 +197,26 @@ def test_delete_user(client, auth_headers, new_user):
...
-    assert is_valid(response, 400) # BAD REQUEST
\ No newline at end of file
+    assert is_valid(response, 400) # BAD REQUEST
+
+
+def test_get_user(client, auth_headers, new_user):
+    """GET '/user' returns a list of all users queried by role"""
+
+    staff_user_response = client.get('/api/user?r=3', headers=auth_headers["admin"])
+    admin_user_response = client.get('/api/user?r=4', headers=auth_headers["admin"])
```
Comment (NickSchimek) [inline tests/integration/test_users.py]:
```suggestion
    staff_user_response = client.get(f'/api/user?r={RoleEnum.STAFF.value}', headers=auth_headers["admin"])
    admin_user_response = client.get(f'/api/user?r={RoleEnum.ADMIN.value}', headers=auth_headers["admin"])
```

File: resources/user.py [hunk 3]
```diff
@@ -204,3 +205,27 @@ def post(self):
...
         return ret, 200
+
+class Users(Resource):
+    @admin_required
+    def get(self):
+
+        role_query = int(request.args["r"])
+
+        if role_query not in range(0, 5):
+            return {"message": "Invalid role"}, 400
```
Comment (NickSchimek) [inline resources/user.py:216]: OK for now, but will break when things change. And the RoleEnum will change soon.

**Review by NickSchimek — CHANGES_REQUESTED** [review]
Thanks for getting this done so fast. 

This assertion `assert all(staff["role"] == 3 for staff in staff_user_response.json["users"])` is technically not asserting anything (replace the 3 with 'anything' to test yourself.), because the query returns an empty response as there is currently no staff seeds. We can remove this and leave the admin tests. The admin tests give me enough confidence that it works correctly. 

Could we replace all the magic numbers with the RoleEnum.

File: resources/user.py (hunk 3 above)
Comment (jls-github) [reply resources/user.py:216]: Ahh okay. How is the RoleEnum changing? Should we just prepare to change this now?

Comment (NickSchimek) [reply resources/user.py:216]: No, let's not take care of it now. I'll create a new issue for that in due time.

**Review by NickSchimek — APPROVED** [review]

**Pull request merged**